src/tutoreact.wiki

```diff
@@ -202,7 +202,7 @@ let display userid_o =
     | _ ->
       [Raw.input ~a:[a_input_type `Text] ()]
   in
-  Lwt.return (messages :: l)
+  Lwt.return (messages @ l)
```

Comment:
Are you sure about this? `messages` seems to be `ul l` (for the `l` in `display_messages`), not a list.